bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -308,6 +309,20 @@ public BigtableSession(BigtableOptions opts) throws IOException {
     initializeResourceLimiter(options);
   }
 
+  private ChannelPool getDataChannelPool() throws IOException {
+    String host = options.getDataHost();
+    int channelCount = options.getChannelCount();
+    if (options.useCachedChannel()) {
+      synchronized (BigtableSession.class) {
+        if (cachedDataChannelPool == null) {
```

Comment:
Might be a good idea to check if the cachedDataChannelPool was constructed with the same host/channelCount